tests/core/Resolver.spec.ts

```diff
@@ -187,5 +189,107 @@ describe('Resolver', () => {
+        createOperationData.signingKeyId,
+        createOperationData.signingPrivateKey
+      );
+      const updateOperation3Data = await OperationGenerator.generateUpdateOperation(
+        createOperationData.createOperation.didUniqueSuffix,
+        createOperationData.nextUpdateRevealValueEncodedString,
+        createOperationData.signingKeyId,
+        createOperationData.signingPrivateKey
+      );
+      const updateOperation1 = OperationGenerator.createAnchoredOperationModelFromOperationModel(updateOperation1Data.updateOperation, 2, 2, 2);
+      const updateOperation2 = OperationGenerator.createAnchoredOperationModelFromOperationModel(updateOperation2Data.updateOperation, 3, 3, 3);
+      const updateOperation3 = OperationGenerator.createAnchoredOperationModelFromOperationModel(updateOperation3Data.updateOperation, 4, 4, 4);
+
+      // Intentionally insert earliest valid recover operation in between the other two operations to test sorting.
+      // Intentionally using the resolver's map construction method to test operations with the same reveal value are placed in the same array.
+      const updateCommitValueToOperationMap: Map<string, AnchoredOperationModel[]>
+        = await (resolver as any).constructCommitValueToOperationLookupMap([updateOperation3, updateOperation1, updateOperation2]);
+      const nextUpdateCommitment = createOperationData.createOperation.delta!.updateCommitment;
+      const updatesWithSameReveal = updateCommitValueToOperationMap.get(nextUpdateCommitment);
+      expect(updatesWithSameReveal).toBeDefined();
+      expect(updatesWithSameReveal!.length).toEqual(3);
+      // const updateCommitValueToOperationMap = new Map<string, AnchoredOperationModel[]>();
+      // updateCommitValueToOperationMap.set(nextUpdateCommitment, [updateOperation3, updateOperation1, updateOperation2]);
+
+      const newDidState: DidState = await (resolver as any).applyUpdateOperations(initialDidState, updateCommitValueToOperationMap);
+
+      // Expecting the new state to contain info of the first recovery operation.
+      expect(newDidState.lastOperationTransactionNumber).toEqual(2);
+      expect(newDidState.nextUpdateCommitmentHash).toEqual(updateOperation1Data.updateOperation.delta!.updateCommitment);
+
+      done();
+    });
+  });
+
+  describe('applyOperation()', () => {
+    it('should not throw error even if an error is thrown internally.', async (done) => {
+      spyOn(operationProcessor, 'apply').and.throwError('any error');
+
+      const createOperationData = await OperationGenerator.generateAnchoredCreateOperation({ transactionTime: 1, transactionNumber: 1, operationIndex: 1 });
+      const initialDidState = await (resolver as any).applyOperation(createOperationData.anchoredOperationModel, undefined);
+
+      // Expecting undefined to be returned instead of error being thrown.
+      expect(initialDidState).toBeUndefined();
+      done();
+    });
   });
```

Comment:
Similar comment as above. I think we are missing tests for the `resolve` where if applying create operation didn't return anything then we return undefined. Similar for recovery/update application etc. Again I would mock the individual `apply...` function instead of constructing complex inputs.

Reply:
Multiple scenario hits the `applyCreateOperation()` returning `undefined`:

* should ignore a deactivate operation of a non-existent did
* should ignore updates to DID that is not created

For this test I did mock the `apply()` method. I do hear the competing principal for writing these tests. So let's chat.